File: frontend/src/api/stats.ts
import { get } from './request'

export interface UsageStats {
  total_tokens: number
  total_requests: number
  avg_latency_ms: number
  success_rate: number
  by_model: {
    model: string
    tokens: number
    requests: number
    cost: number
  }[]
  by_day: {
    date: string
    tokens: number
    requests: number
  }[]
}

export const getUsageStats = (params: {
  start_date: string
  end_date: string
  group_by?: string
}) => get<UsageStats>('/stats/usage', { params })

export interface AdminStats {
  total_tokens: number
  total_requests: number
  by_user: {
    user_id: string
    username: string
    tokens: number
  }[]
  by_provider: {
    provider: string
    tokens: number
  }[]
}

export const getAdminStats = (params: {
  start_date: string
  end_date: string
  group_by?: string
}) => get<AdminStats>('/admin/stats/usage', { params })

File: backend/app/schemas/stats.py
"""
用量统计相关Schema
"""
from pydantic import BaseModel
from typing import List, Optional


class UsageStats(BaseModel):
    """用量统计响应"""
    total_tokens: int = 0
    total_requests: int = 0
    avg_latency_ms: int = 0
    success_rate: float = 100.0


class ModelUsage(BaseModel):
    """模型使用统计"""
    model: str
    tokens: int
    requests: int
    cost: float = 0.0


class DailyUsage(BaseModel):
    """每日使用统计"""
    date: str
    tokens: int
    requests: int


class UsageStatsResponse(UsageStats):
    """完整用量统计响应"""
    by_model: List[ModelUsage] = []
    by_day: List[DailyUsage] = []


class UsageQuery(BaseModel):
    """用量查询参数"""
    start_date: Optional[str] = None
    end_date: Optional[str] = None
    model: Optional[str] = None
